### ISSUE TITLE
patch error with skeleton pypi on Windows 7 64-bit

### ISSUE THREAD

**Issue by asmeurer** [conda/conda-build#119]
Reported on the conda mailing list. 

```

C:\gca\temp>conda skeleton pypi pymarc
Using url https://pypi.python.org/packages/source/p/pymarc/pymarc-3.0.1.tar.gz (131783) for pymarc.
Downloading pymarc (use --no-download to skip this step)
Unpacking pymarc...
done
working in c:\users\gaborca\appdata\local\temp\tmpvjyhclconda_skeleton_pymarc
Fetching package metadata: ....
Solving package specifications: .
Applying patch: u'c:\\users\\gaborca\\appdata\\local\\temp\\tmpvjyhclconda_skeleton_pymarc\\pypi-distutils.patch'
69 unparsed bytes left at the end of stream
file 1/1:        core.py
 hunk no.1 doesn't match source file at line 165
  expected:
  actual  :             if DEBUG:
source file is different - core.py
Traceback (most recent call last):
  File "C:\gca\program_files\Miniconda\Scripts\conda-skeleton-script.py", line 4, in <module>
    sys.exit(main())
  File "C:\gca\program_files\Miniconda\lib\site-packages\conda_build\main_skeleton.py", line 124, in main
    args.func(args, p)
  File "C:\gca\program_files\Miniconda\lib\site-packages\conda_build\main_skeleton.py", line 135, in execute
    pypi.main(args, parser)
  File "C:\gca\program_files\Miniconda\lib\site-packages\conda_build\pypi.py", line 314, in main
    run_setuppy(src_dir, tempdir, args)
  File "C:\gca\program_files\Miniconda\lib\site-packages\conda_build\pypi.py", line 485, in run_setuppy
    apply_patch(join(stdlib_dir, 'distutils'), patch)
  File "C:\gca\program_files\Miniconda\lib\site-packages\conda_build\source.py", line 200, in apply_patch
    check_call([patch, '-p0', '-i', path], cwd=src_dir)
  File "C:\gca\program_files\Miniconda\lib\subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '[u'C:\\gca\\program_files\\Miniconda\\Scripts\\patch.bat', u'-p0', u'-i', u'c:\\users\\gaborca\\appdata\\local\\temp\\tm
pvjyhclconda_skeleton_pymarc\\pypi-distutils.patch']' returned non-zero exit status -1

```

**Comment by plakdawa** [comment]
I have exactly the same issue listed originally on this thread when trying to create a skeleton for wtforms.

Is there a known workaround for this?

**Comment by pseudonymcp** [comment]
Me too, I also have exactly the same issue. I installed conda-build and patch and get the same error

```
Applying patch: 'c:\\users\\pseudo\\appdata\\local\\temp\\tmpiodhraconda_skeleton_Pyped-1.4.tar.gz\\pypi-distutils.patch'
69 unparsed bytes left at the end of stream
file 1/1:        core.py
 hunk no.1 doesn't match source file at line 165
  expected:
  actual  :                 raise SystemExit, "error: " + str(msg)
source file is different - core.py
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda-build/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "C:\Users\pseudo\Anaconda\Scripts\conda-skeleton-script.py", line 4, in <module>
    sys.exit(main())
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\main_skeleton.py", line 188, in main
    args_func(args, p)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\main_build.py", line 321, in args_func
    args.func(args, p)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\main_skeleton.py", line 202, in execute
    pypi.main(args, parser)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\pypi.py", line 376, in main
    get_package_metadata(args, package, d, data)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\pypi.py", line 432, in get_package_metadata
    run_setuppy(src_dir, tempdir, args)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\pypi.py", line 690, in run_setuppy
    apply_patch(join(stdlib_dir, 'distutils'), patch)
  File "C:\Users\pseudo\Anaconda\lib\site-packages\conda_build\source.py", line 235, in apply_patch
    check_call([patch, '-p0', '-i', path], cwd=src_dir)
  File "C:\Users\pseudo\Anaconda\lib\subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['C:\\Users\\pseudo\\Anaconda\\Scripts\\patch.bat', '-p0', '-i', 'c:\\users\\pseudo\\ap
pdata\\local\\temp\\tmpiodhraconda_skeleton_Pyped-1.4.tar.gz\\pypi-distutils.patch']' returned non-zero exit status -1
```

Conda info:
```
Current conda install:

             platform : win-32
        conda version : 3.10.0
  conda-build version : 1.11.0
       python version : 2.7.8.final.0
     requests version : 2.5.3
     root environment : C:\Users\pseudo\Anaconda  (writable)
  default environment : C:\Users\pseudo\Anaconda
     envs directories : C:\Users\pseudo\Anaconda\envs
        package cache : C:\Users\pseudo\Anaconda\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-32/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-32/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
```
I'm working on Win7 64-bit with a 32-bit Anaconda installation.

**Comment by asmeurer** [comment]
We need to replace our patch package with the one from http://gnuwin32.sourceforge.net/packages/patch.htm. As a workaround, you can `conda remove patch` and install `patch` from that website.

**Comment by cruser42** [comment]
Hi,

That fix worked perfectly for me.

Best,
-Aleksey

**Comment by remram44** [comment]
This still happens

#478
#476

It is currently impossible to build packages on Windows (either 32 or 64) without major hassle. Please, do something.

**Comment by remram44** [comment]
Just remove the conda "patch" package if it doesn't work for your patches, at least we'll know to look somewhere else.

**Comment by LeslieK** [comment]
Your link to the patch URL is broken.
How do we install pypi packages when conda is broken for windows 7 64?
conda version 3.14.1
python: 3.4.3.final.0
My patch error output:

conda skeleton pypi <package>
fails with the error:
Did not find 'patch' in: C:\Anaconda3\envs....
You can install 'patch' ...

I am using python 3.4 and windows 7 64-bit.
conda install patch
Hint: the following packages conflict with each other:
- patch
- python 3.4*

**Comment by remram44** [comment]
Just get a "patch" binary from somewhere else, like MSYS. The conda "patch" package wouldn't work anyway, that's what this issue describes.

**Comment by LeslieK** [comment]
Could you include a link to where I get the "patch" file? I'm new at this. I assume I download it and put int somewhere in my PATH. Not familiar with MSYS. Thanks.

**Comment by LeslieK** [comment]
Got the patch by using 
mingw-get install msys-patch

This should be the message that is displayed with the conda skeleton pip <package> command when the patch is not found. 
Error:
    You can install 'patch' using apt-get, yum (Linux), Xcode (MacOSX),
    or conda, **cygwin** (Windows),

Better error message is to put in the exact command like:
mingw-get install msys-patch

Solution: a better error message

**Error:**
You can install 'patch' using apt-get, yum (Linux), Xcode (MacOSX), conda,
**mingw-get install msys-patch** (Windows)

**Comment by remram44** [comment]
MSYS and MinGW are different things from Cygwin. IMHO there's no need to list every method of getting the file, but **the conda method should definitely be removed since the binary will not work for conda's purposes**.

**Comment by msarahan** [comment]
I believe these issues are fixed with the patch package, version 2.5.9, build 1.  This bundles the GnuWin32 patch executable with a manifest that tells Windows to not require elevation when running patch.exe.

If you have patch.bat in your scripts folder, please remove it.  All tracebacks using the proper program should show "patch.exe" not "patch.bat"

Closing this issue.  If there are further issues with patch, please post new issues with tracebacks.

**Comment by remram44** [comment]
Confirmed. Thanks!

Had a quick look at open issues and probably a lot of them should now be closed.
